Sources/Swift/Persistence/SentryScopeContextPersistentStore.swift

```diff
@@ -0,0 +1,83 @@
+@_implementationOnly import _SentryPrivate
+
+@objcMembers
+class SentryScopeContextPersistentStore: NSObject {
+    private let fileManager: SentryFileManager
+
+    init(fileManager: SentryFileManager) {
+        self.fileManager = fileManager
+    }
+
+    // MARK: - Context
+
+    func readPreviousContextFromDisk() -> [String: [String: Any]]? {
+        let fm = FileManager.default
+        guard fm.fileExists(atPath: contextFileURL.path) else {
+            return nil
+        }
+        do {
+            let data = try Data(contentsOf: contextFileURL)
+            return decodeContext(from: data)
+        } catch {
+            SentryLog.error("Failed to read context data from file at path: \(contextFileURL), reason: \(error)")
+            return nil
+        }
+    }
+
+    func writeContextToDisk(context: [String: [String: Any]]) {
+        guard let data = encode(context: context) else {
+            return
+        }
+        do {
+            try data.write(to: contextFileURL, options: .atomic)
```

Comment:
Where does the path to this file get created? If the sdk is started with a custom cache path that's a deeply nested path where none of the directories are on disk yet, does this still work? Because `-[SentryFileManager writeData:toPath:]` creates such directory paths on the fly, but since we're not calling that here (which I'd recommend anyways) I don't think this will work in the general case.

Reply:
`contextFileURL` is a an alias/wrapper for `fileManager.contextFilePath` (see end of file). The `contextFilePath` is initialized in `SentryFileManager` like every other path based on the `sentryPath` which is backed by the `options.cacheDirectoryPath` and dsn hash.

I'll see if I can reuse logic of SentryFileManager (if it's not too specific for other use cases, I do not want to pollute new logic with old one).